collections-lists/FastestWayToCheckIfListIsOrdered/FastestWayToCheckIfListIsOrdered/ListOrderValidator.cs

```diff
@@ -0,0 +1,202 @@
+﻿using System.Runtime.InteropServices;
+
+namespace FastestWayToCheckIfListIsOrdered;
+
+public class ListOrderValidator
+{
+	public static bool IsOrderedUsingForLoop<T>(IList<T> list, IComparer<T>? comparer = default)
+	{
+		if (list.Count <= 1) return true;
+
+		comparer ??= Comparer<T>.Default;
+		var length = list.Count;
+		for (var i = 1; i < length; i++)
+		{
+			if (comparer.Compare(list[i - 1], list[i]) > 0) return false;
+		}
+
+		return true;
+	}
+
+	public static bool IsOrderedUsingArraySort<T>(IList<T> list, T[] array, IComparer<T>? comparer = default)
+	{
+		if (list.Count <= 1) return true;
+
+		comparer ??= Comparer<T>.Default;
+
+		var length = list.Count;
+
+		list.CopyTo(array, 0);
+		Array.Sort(array, 0, length, comparer);
+
+		for (var i = 0; i < length; i++)
+		{
+			if (comparer.Compare(list[i], array[i]) != 0) return false;
+		}
+
+		return true;
+	}
+
+	public static bool IsOrderedUsingSpans<T>(List<T> list, IComparer<T>? comparer = default)
+	{
+		comparer ??= Comparer<T>.Default;
+		var span = CollectionsMarshal.AsSpan(list);
+
+		var length = span.Length;
```

Comment:
For consistency, we should also have the "bail early" check of `list.Count <= 1` here.